File: runtimes/swiftui/Sources/StyleConverterRuntime/StyleEngine/background/BackgroundRepeatConfig.swift
//
//  BackgroundRepeatConfig.swift
//  StyleEngine/background — Phase 4.
//
//  CSS `background-repeat` per layer. IR shape:
//    ["repeat"|"no-repeat"|"space"|"round"]  OR
//    [{x: <kw>, y: <kw>}]
//  We parse into a simple per-layer tuple. SwiftUI has no gradient
//  repeat primitive — the applier is a stub.
//

import Foundation

// One layer's repeat value — the X and Y axes may differ.
struct BackgroundRepeatLayer: Equatable {
    // Normalised lower-case keyword: repeat / no-repeat / space / round.
    var x: String
    // Same axis vocabulary.
    var y: String
}

struct BackgroundRepeatConfig: Equatable {
    // One entry per background-image layer.
    var layers: [BackgroundRepeatLayer] = []
    // Short-circuit for the applier.
    var hasAny: Bool { !layers.isEmpty }
}

File: runtimes/web/src/engine/background/BackgroundRepeatConfig.ts
// BackgroundRepeatConfig.ts — typed record for CSS `background-repeat`.
// IR is an array: each entry is either a keyword string
// ('repeat'|'no-repeat'|'space'|'round'|'repeat-x'|'repeat-y') or an
// axis-pair object { x: keyword, y: keyword }.

// One entry per layer — pre-rendered CSS fragment.
export interface BackgroundRepeatLayer {
  css: string;                                                        // CSS fragment for one layer
}

// Config holder.
export interface BackgroundRepeatConfig {
  layers: BackgroundRepeatLayer[];                                    // empty = unset
}

// IR property type.
export const BACKGROUND_REPEAT_PROPERTY_TYPE = 'BackgroundRepeat' as const;
export type BackgroundRepeatPropertyType = typeof BACKGROUND_REPEAT_PROPERTY_TYPE;
